### PR TITLE
turn on neural_speed by default

### PR DESCRIPTION
**Pull request by yufenglee**
### Description

the crash caused by the neural_speed turns out to be a very corn case. Turn it on by default.

### CODE REVIEW

**Review by yihonglyu — APPROVED** [review]

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -88,7 +88,7 @@ option(onnxruntime_USE_QNN "Build with QNN support" OFF)
 option(onnxruntime_USE_SNPE "Build with SNPE support" OFF)
 option(onnxruntime_USE_RKNPU "Build with RKNPU support" OFF)
 option(onnxruntime_USE_DNNL "Build with DNNL support" OFF)
-option(onnxruntime_USE_NEURAL_SPEED "Build with Neural Speed support" OFF)
+option(onnxruntime_USE_NEURAL_SPEED "Build with Neural Speed support" ON)
```
[the remark this answers is not in the document]
Comment (yufenglee) [reply cmake/CMakeLists.txt:91]: what is the issue with xbyak's license?

**Comment by snnn** [comment]
In my opinion the problem still should be fixed, especially we see the fix is there but for what reason we refuse doing so? Windows and Office are deployed to billions of devices. So definitely some people will hit the issue.  Even if they only represent a small percentage of the total population, we will still see a huge number of crash reports. 
Besides, xbyak is also a new dependency which may also have compatibility issues. This is the first time we introduce a JIT engine in the core framework and the code will be deployed billions of devices. Should we be more careful on that? Anyone peeked into xbyak's code? xbyak does a lot of non-trivial things. I would think this is the biggest change we ever had in ONNX Runtime in the past 6 years.

File: cmake/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (snnn) [reply cmake/CMakeLists.txt:91]: Intel copied their code and didn't disclose that. It is copyright infringement that can be sued.

**Comment by yufenglee** [comment]
> In my opinion the problem still should be fixed, especially we see the fix is there but for what reason we refuse doing so? Windows and Office are deployed to billions of devices. So definitely some people will hit the issue. Even if they only represent a small percentage of the total population, we will still see a huge number of crash reports. Besides, xbyak is also a new dependency which may also have compatibility issues. This is the first time we introduce a JIT engine in the core framework and the code will be deployed billions of devices. Should we be more careful on that? Anyone peeked into xbyak's code? xbyak does a lot of non-trivial things. I would think this is the biggest change we ever had in ONNX Runtime in the past 6 years.

the fix is in progress. At the same time, we don't want user to experience very slow experience for the corner case.

File: cmake/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (yufenglee) [reply cmake/CMakeLists.txt:91]: We have included their copyright in our code: https://github.com/microsoft/onnxruntime/blob/aec2389ad0463d218b8cf3b1e245d4c34e98364a/ThirdPartyNotices.txt#L1010

**Comment by snnn** [comment]
Please add xbyak's license to https://github.com/microsoft/onnxruntime/blob/main/ThirdPartyNotices.txt

**Review by snnn — APPROVED** [review]

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**